mobile/native/src/ln_dlc/mod.rs

```diff
@@ -587,37 +591,53 @@ fn derive_trades_from_filled_orders() -> Result<Vec<WalletHistoryItem>> {
                         open_order.direction,
                     )?;
 
+                    let opening_price = Decimal::from_f32(opening_price).expect("to fit into f32");
+                    let open_position_fee =
+                        order_matching_fee_taker(open_order.quantity, opening_price).to_sat()
+                            as i64;
+                    let pnl_after_open_position_fee = pnl_before_fees + open_position_fee;
+
```

Comment:
I do not understand: why do you subtract the `open_position_fee` from the PnL?

Reply:
Because the pnl is calculated from the margin that does not include the fees, but the user actually put up the margin with fees. So if we are calculating the pnl it always includes the open position fee. Since we should that fee separately I am deducting it from the pnl.

Reply:
I'm not sure this is correct. In the pnl calculation we fill in the `opening_price`, the `closing_price` and the `leverage`. We do not put in the margin the user has to put up but we calculate the margin from the price and the leverage which does not include the fee.

Reply:
As discussed in a call. Here is a [link](https://docs.google.com/spreadsheets/d/1VD-OnAuYzTYxBerkGifUEWt-XYcXGCl58SCMXXyv-_g/edit?usp=sharing) to our calculations